### DISCUSSION TITLE
有关minitouch的下位替代

### DISCUSSION THREAD

**Discussion by aa889788** [MaaAssistantArknights/MaaAssistantArknights#2917] [Ideas]
因为minitouch依赖实际存在的设备来转发事件，在一些没有实体设备的模拟器上（比如redroid）会无法工作。因为担心你们的开发重心会切换到minitouch上，所以我用java写了一个和minitouch命令兼容的包，经过在我本地设备上测试，可以替代minitouch工作，并且因为是直接向InputManager发送事件，所以无需像minitouch一样考虑分辨率不同的scale和rotate，和adb tap一样点对点即可。
因为我的两个设备都是android 12，没有测试过在不同版本上的兼容性，发出来是想开发组考虑一下是否可能替代minitouch，以及有不同版本设备的人可以帮忙测试一下兼容性（不过关键的触控模拟部分是从scrcpy KANG来的，应该不会有问题）
```
adb push touch.zip /data/local/tmp/touch.jar
adb shell "export CLASSPATH=/data/local/tmp/touch.jar; app_process /data/local/tmp com.shxyke.touchevent.App"
```
[touchevent](https://github.com/aa889788/touchevent)
[touch.zip](https://github.com/MaaAssistantArknights/MaaAssistantArknights/files/10098230/touch.zip)

**Comment by ABA2396** [comment]
连接设置里有个兼容性的触控模式，那个用的还是原来的点击吧

**Reply by aa889788** [reply]
是的，不过adb的点击可定制性和效率不太行，前段时间研究了一些scrcpy的实现，就写了这个库，而且因为是Java写的，也不需要像minitouch那样考虑abi之类的

**Comment by MistEO** [comment]
已整合并更名为 MaaTouch！🤗
